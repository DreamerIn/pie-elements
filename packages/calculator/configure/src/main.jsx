import React from 'react';
import ReactDOM from 'react-dom';
import Calculator from '@pie-framework/material-ui-calculator';
import { InputRadio, TwoChoice } from '@pie-lib/config-ui';

export default class Main extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      twoChoice: '',
      model: props.model
    };
  }

  render() {
    const { basic, scientific } = this.state;
    return (
      <div>
        <TwoChoice
          header="Choose Calculator Type"
          value={this.state.twoChoice}
          onChange={twoChoice => this.setState({ twoChoice })}
          one={{ label: 'Basic', value: 'basic' }}
          two={{ label: 'Scientific', value: 'scientific' }} />
          {this.state.twoChoice && <Calculator mode={this.state.twoChoice} />}
      </div>
    );
  }
}