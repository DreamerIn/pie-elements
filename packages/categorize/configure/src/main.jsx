import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { Tabs } from '@pie-lib/config-ui';
import Design from './design';
import Scoring from './scoring';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    imageSupport: PropTypes.object
  };

  static defaultProps = {};

  changeScoring = scoring => {
    const { onChange, model } = this.props;
    model.scoring = scoring;
    onChange(model);
  };

  render() {
    const { classes, className, model, onChange } = this.props;

    if (!model.scoring) {
      model.scoring = {
        weighting: {
          enabled: false,
          rules: []
        },
        partial: {
          enabled: false,
          rules: []
        }
      };
    }

    return (
      <div className={classNames(classes.main, className)}>
        <Tabs>
          <Design
            imageSupport={this.props.imageSupport}
            title="Design"
            model={model}
            onChange={onChange}
          />
          <Scoring
            title="Scoring"
            scoring={model.scoring}
            categories={model.categories}
            correctResponse={model.correctResponse}
            onChange={this.changeScoring}
          />
        </Tabs>
      </div>
    );
  }
}
const styles = theme => ({
  main: {}
});

const StyledMain = withStyles(styles)(Main);

class Stateful extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object.isRequired,
    imageSupport: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      model: props.model
    };
  }

  componentWillReceiveProps(props) {
    this.setState({ model: props.model });
  }

  onChange = model => {
    this.setState({ model }, () => {
      this.props.onChange(this.state.model);
    });
  };

  render() {
    return (
      <StyledMain
        imageSupport={this.props.imageSupport}
        model={this.state.model}
        onChange={this.onChange}
      />
    );
  }
}

export default Stateful;
