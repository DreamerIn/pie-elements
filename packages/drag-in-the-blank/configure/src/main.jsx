import React from 'react';
import PropTypes from 'prop-types';
import EditableHtml, { ALL_PLUGINS } from '@pie-lib/editable-html';
import {
  InputContainer,
  layout,
  settings
} from '@pie-lib/config-ui';
import { withDragContext } from '@pie-lib/drag';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Choices from './choices';
const { toggle, Panel } = settings;

const styles = theme => ({
  promptHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
  markup: {
    minHeight: '235px',
    paddingTop: theme.spacing.unit * 2,
    width: '100%',
    '& [data-slate-editor="true"]': {
      minHeight: '235px'
    }
  },
  design: {
    paddingTop: theme.spacing.unit * 3
  },
  choiceConfiguration: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  switchElement: {
    justifyContent: 'space-between',
    margin: 0
  },
  addButton: {
    float: 'right'
  },
  text: {
    fontFamily: 'Cerebri Sans',
    fontSize: '16px',
    lineHeight: '19px',
    color: '#495B8F'
  }
});

export class Main extends React.Component {
  static propTypes = {
    configuration: PropTypes.object.isRequired,
    model: PropTypes.object.isRequired,
    disableSidePanel: PropTypes.bool,
    onModelChanged: PropTypes.func.isRequired,
    onConfigurationChanged: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  onModelChange = newVal => {
    this.props.onModelChanged({
      ...this.props.model,
      ...newVal
    });
  };

  onPromptChanged = prompt => {
    this.props.onModelChanged({
      ...this.props.model,
      prompt
    });
  };

  onMarkupChanged = slateMarkup => {
    this.props.onModelChanged({
      ...this.props.model,
      slateMarkup
    });
  };

  onResponsesChanged = choices => {
    this.props.onModelChanged({
      ...this.props.model,
      choices
    });
  };

  render() {
    const {
      classes,
      model,
      configuration,
      onConfigurationChanged,
      imageSupport
    } = this.props;
    const {
      duplicates,
      prompt,
      partialScoring,
      lockChoiceOrder
    } = configuration;

    return (
      <div className={classes.design}>
        <layout.ConfigLayout
          settings={
            <Panel
              model={model}
              configuration={configuration}
              onChangeModel={model => this.onModelChange(model)}
              onChangeConfiguration={configuration => onConfigurationChanged(configuration, true)}
              groups={{
                'Item Type': {
                  partialScoring: partialScoring.settings &&
                  toggle(partialScoring.label),
                  duplicates: duplicates.settings && toggle(duplicates.label)
                },
                'Properties': {
                  lockChoiceOrder: lockChoiceOrder.settings &&
                  toggle(lockChoiceOrder.label)
                },
              }}
            />
          }
        >
          <div>
            {prompt.settings && (
              <InputContainer
                label={prompt.label}
                className={classes.promptHolder}
              >
                <EditableHtml
                  className={classes.prompt}
                  markup={model.prompt}
                  onChange={this.onPromptChanged}
                  imageSupport={imageSupport}
                  nonEmpty={!prompt.settings}
                  disableUnderline
                />
              </InputContainer>
            )}
            <Typography className={classes.text}>
              Define Template, Choices, and Correct Responses
            </Typography>
            <EditableHtml
              activePlugins={ALL_PLUGINS}
              toolbarOpts={{
                position: 'top',
                alwaysVisible: true
              }}
              responseAreaProps={{
                type: 'drag-in-the-blank',
                options: {
                  duplicates: model.duplicates
                }
              }}
              className={classes.markup}
              markup={model.slateMarkup}
              onChange={this.onMarkupChanged}
              imageSupport={imageSupport}
              nonEmpty={!prompt.settings}
              disableUnderline
            />
            <Choices
              model={model}
              duplicates={model.duplicates}
              onChange={this.onResponsesChanged}
            />
          </div>
        </layout.ConfigLayout>
      </div>
    );
  }
}

const Styled = withStyles(styles)(Main);

export default withDragContext(Styled);