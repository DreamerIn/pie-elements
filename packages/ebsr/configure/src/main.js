import { ModelUpdatedEvent } from '@pie-framework/pie-configure-events';
import MultipleChoiceConfigure from '@pie-element/multiple-choice/configure/lib';

import debug from 'debug';

const log = debug('pie-elements:ebsr:configure');

export default class Main extends HTMLElement {
  static defineMultipleChoice() {
    if(!customElements.get('multiple-choice-configure')){
      customElements.define('multiple-choice-configure', MultipleChoiceConfigure);
    }
  }

  static getParts = () => ({
    partA: document.getElementById('part-a-configure'),
    partB: document.getElementById('part-b-configure'),
  });

  constructor() {
    super();

    Main.defineMultipleChoice();
  }

  set model(m) {
    this._model = m;

    this._render();
    this.setupModel();
  }

  setupModel() {
    const { partA, partB } = Main.getParts();

    if (partA && partB) {
      this.setupPart(partA, 'partA');
      this.setupPart(partB, 'partB');
    }
  }

  setupPart(part, key) {
    this.initiatePart(part, key);
    this.captureModelUpdated(part, key);
  }

  initiatePart(part, key) {
    part.model = this._model[key];

    this.dispatchModelChanged();
  }

  dispatchModelChanged(reset = false) {
    log('[onModelChanged]: ', this._model);

    this.dispatchEvent(new ModelUpdatedEvent(this._model, reset));
  }

  captureModelUpdated(part, key) {
    const self = this;

    part.addEventListener('model.updated', event => {
      event.stopImmediatePropagation();
      self.repackModel(event.update, key);
    });
  }

  repackModel(partModel, key) {
    this._model =  {
      ...this._model,
      [key]: partModel
    };

    this.dispatchModelChanged();
  }

  _render() {
    this.innerHTML = `
      <div>
        <multiple-choice-configure id="part-a-configure"></multiple-choice-configure>
        <multiple-choice-configure id="part-b-configure"></multiple-choice-configure>
      </div>
    `;
  }
}
