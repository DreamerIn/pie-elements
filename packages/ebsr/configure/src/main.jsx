import React from 'react';
import PropTypes from 'prop-types';

import MultipleChoiceConfigure from '@pie-element/multiple-choice/configure/lib';

export default class Root extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    onModelChanged: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    if(!customElements.get('multiple-choice-configure')){
      customElements.define('multiple-choice-configure', MultipleChoiceConfigure);
    }

    this.state = {
      model: props.model,
    };
  }

  componentDidMount() {
    const { partA, partB } = this;

    customElements.whenDefined('multiple-choice-configure').then(() => {
      this.configurePart(partA, 'partA');
      this.configurePart(partB, 'partB');
    });
  }

  configurePart(part, key) {
    this.updatePart(part, key, true);
    this.handleModelUpdateListener(part, key);
  }

  updatePart(part, key, setModel) {
    const { onModelChanged } = this.props;
    const { model } = this.state;

    const { keyMode } = model;
    const { choices, choiceMode, labelMode, prompt } = model[key];

    const partModel = {
      prompt,
      choiceMode,
      keyMode,
      choices,
      labelMode,
      // hardcoded for now
      lockChoiceOrder: false,
      sequentialLabel: false
    };

    if (setModel) {
      part.model = partModel;
    }

    const repackedModel = {
      ...model,
      [key]: partModel,
    };

    onModelChanged(repackedModel);
  }

  handleModelUpdateListener(part, key) {
    const self = this;

    part.addEventListener('model.updated', (model) => {
      self.updatePart(model.update, key);
    })
  }

  render() {
    return (
      <div>
        <multiple-choice-configure ref={ref => { this.partA = ref }} />
        <multiple-choice-configure ref={ref => { this.partB = ref }} />
      </div>
    );
  }
}
