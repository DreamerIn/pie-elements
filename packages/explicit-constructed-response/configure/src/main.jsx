import React from 'react';
import PropTypes from 'prop-types';
import EditableHtml, { ALL_PLUGINS } from '@pie-lib/editable-html';
import {
  InputContainer,
  layout
} from '@pie-lib/config-ui';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import AlternateResponses from './alternatResponses';

const styles = theme => ({
  promptHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
  markup: {
    minHeight: '235px',
    paddingTop: theme.spacing.unit * 2,
    width: '100%',
    '& [data-slate-editor="true"]': {
      minHeight: '235px'
    }
  },
  design: {
    paddingTop: theme.spacing.unit * 3
  },
  choiceConfiguration: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  switchElement: {
    justifyContent: 'space-between',
    margin: 0
  },
  addButton: {
    float: 'right'
  },
  text: {
    fontFamily: 'Cerebri Sans',
    fontSize: '16px',
    lineHeight: '19px',
    color: '#495B8F'
  }
});

export class Main extends React.Component {
  static propTypes = {
    configuration: PropTypes.object.isRequired,
    model: PropTypes.object.isRequired,
    disableSidePanel: PropTypes.bool,
    onModelChanged: PropTypes.func.isRequired,
    onConfigurationChanged: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  onPromptChanged = prompt => {
    this.props.onModelChanged({
      ...this.props.model,
      prompt
    });
  };

  onMarkupChanged = slateMarkup => {
    this.props.onModelChanged({
      ...this.props.model,
      slateMarkup
    });
  };

  render() {
    const {
      classes,
      model,
      configuration,
      imageSupport
    } = this.props;
    const {
      prompt
    } = configuration;

    return (
      <div className={classes.design}>
        <layout.ConfigLayout>
          <div>
            {prompt.settings && (
              <InputContainer
                label={prompt.label}
                className={classes.promptHolder}
              >
                <EditableHtml
                  className={classes.prompt}
                  markup={model.prompt}
                  onChange={this.onPromptChanged}
                  imageSupport={imageSupport}
                  nonEmpty={!prompt.settings}
                  disableUnderline
                />
              </InputContainer>
            )}
            <Typography className={classes.text}>
              Define Template, Choices, and Correct Responses
            </Typography>
            <InputContainer
              className={classes.promptHolder}
            >
              <EditableHtml
                activePlugins={ALL_PLUGINS}
                toolbarOpts={{
                  position: 'top',
                  alwaysVisible: true
                }}
                responseAreaType="explicit-constructed-response"
                className={classes.markup}
                markup={model.slateMarkup}
                onChange={this.onMarkupChanged}
                imageSupport={imageSupport}
                nonEmpty={!prompt.settings}
                disableUnderline
              />
            </InputContainer>
            <Typography className={classes.text}>
              Define Alternates
            </Typography>
            <AlternateResponses
              model={model}
            />
          </div>
        </layout.ConfigLayout>
      </div>
    );
  }
}

const Styled = withStyles(styles)(Main);

export default Styled;