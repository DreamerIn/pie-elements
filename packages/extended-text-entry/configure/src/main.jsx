import React from 'react';
import {
  NumberTextField,
  InputCheckbox,
  FeedbackSelector,
  InputContainer
} from '@pie-lib/config-ui';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import EditableHtml from '@pie-lib/editable-html';

const defaultFeedback = {
  type: 'default',
  default: 'Your answer has been submitted'
};

export class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      setDimensions: true
    };
    this.changeWidth = this.change('width');
    this.changeHeight = this.change('height');
  }

  change = key => (event, v) => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ [key]: v });
    onChange(model);
  };

  onPromptChange = markup => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ prompt: markup });
    onChange(model);
  };

  applyUpdate(update) {
    const { model } = this.props;
    const out = Object.assign({}, model, update);
    return out;
  }

  toggleMath = event => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ showMathInput: event.target.checked });
    onChange(model);
  };

  changeFeedback = feedback => {
    const { model, onChange } = this.props;
    const update = { ...model, feedback };
    onChange(update);
  };

  render() {
    const { model, classes } = this.props;
    return (
      <div>
        <Typography className={classes.header} variant="subheading">
          Display
        </Typography>
        <NumberTextField
          label="Width (px)"
          disabled={!this.state.setDimensions}
          value={parseInt(model.width)}
          min={100}
          max={500}
          onChange={this.changeWidth}
        />
        <NumberTextField
          label="Height (px)"
          disabled={!this.state.setDimensions}
          value={parseInt(model.height)}
          min={100}
          max={500}
          onChange={this.changeHeight}
        />
        <br />
        <InputCheckbox
          label={'Student responses can include math notation'}
          onChange={this.toggleMath}
          checked={!!model.showMathInput}
        />
        <br />
        <InputContainer label="Prompt" className={classes.promptContainer}>
          <EditableHtml markup={model.prompt} onChange={this.onPromptChange} />
        </InputContainer>

        <Typography className={classes.header} variant="subheading">
          Feedback
        </Typography>

        <FeedbackSelector
          label="When submitted, show"
          feedback={model.feedback || defaultFeedback}
          onChange={this.changeFeedback}
        />
      </div>
    );
  }
}
export default withStyles(theme => ({
  header: {
    paddingBottom: theme.spacing.unit
  },
  promptContainer: {
    paddingTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2,
    width: '100%'
  },
  promptInput: {
    width: '100%',
    marginBottom: theme.spacing.unit
  }
}))(Main);
