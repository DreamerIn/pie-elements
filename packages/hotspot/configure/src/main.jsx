import React from 'react';
import PropTypes from 'prop-types';
import { ConfigLayout } from '@pie-lib/config-ui';
import { withStyles } from '@material-ui/core/styles';
import Switch from '@material-ui/core/Switch';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import Hotspot from './hotspot';

const styles = theme => ({
  promptHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
  design: {
    paddingTop: theme.spacing.unit * 3
  },
  choiceConfiguration: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  switchElement: {
    justifyContent: 'space-between',
    margin: 0
  },
  addButton: {
    float: 'right'
  }
});

const getSideMenuItems = (props) => {
  const { classes, configure, onPartialScoringChanged, onMultipleCorrectChanged } = props;

  const { partialScoring, settingsPartialScoring, multipleCorrect, settingsMultipleCorrect } = configure;

  return [
    {
      items: [
        settingsMultipleCorrect && <FormControlLabel
          key={3}
          classes={{
            root: classes.switchElement
          }}
          control={
            <Switch
              checked={multipleCorrect}
              onChange={onMultipleCorrectChanged}
              value="checkedA"
            />
          }
          label="Multiple Correct Responses"
          labelPlacement="start"
        />,
        settingsPartialScoring && <FormControlLabel
          key={3}
          classes={{
            root: classes.switchElement
          }}
          control={
            <Switch
              checked={partialScoring}
              onChange={onPartialScoringChanged}
              value="checkedA"
            />
          }
          label="Allow Partial Scoring"
          labelPlacement="start"
        />,
      ]
    }
  ];
};

const Design = withStyles(styles)(props => {
  const {
    classes,
    disableSidePanel,
  } = props;

  return (
    <div className={classes.design}>
      <ConfigLayout
        sideMenuItems={getSideMenuItems(props)}
        regularItems={
          <div>
            <Hotspot />
          </div>
        }
        disableSidePanel={disableSidePanel}
      />
    </div>
  );
});

export class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    disableSidePanel: PropTypes.bool,
    onPromptChanged: PropTypes.func.isRequired,
    onPartialScoringChanged: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  render() {
    return (
      <Design {...this.props} />
    );
  }
}

const Styled = withStyles(styles)(Main);

export default Styled;

