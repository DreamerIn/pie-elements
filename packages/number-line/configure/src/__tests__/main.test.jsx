/* eslint-disable react/display-name */
import { shallow } from 'enzyme';
import React from 'react';
import { Main } from '../main';
import {
  model as defaultModel,
  configuration as defaultConfig
} from '../defaults';

jest.mock('@pie-lib/config-ui', () => ({
  FormSection: () => <div />,
  FeedbackConfig: () => <div />,
  InputCheckbox: () => <div />
}));

describe('main', () => {
  let w;
  let onChange = jest.fn();
  const wrapper = extras => {
    const defaults = {
      classes: {},
      className: 'className',
      onChange,
      model: { correctResponse: [], graph: { ...defaultModel } },
      configuration: { ...defaultConfig }
    };
    const props = { ...defaults, ...extras };
    return shallow(<Main {...props} />);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
  describe('logic', () => {
    const fn = (fnName, args, expected) => {
      describe(fnName, () => {
        it(`(${args.map(a => JSON.stringify(a)).join(',')})`, () => {
          const w = wrapper();
          const i = w.instance();
          i[fnName].apply(w.instance(), args);
          expect(onChange).toHaveBeenCalledWith(expected);
        });
      });
    };

    fn('graphChange', [{ foo: true }], {
      graph: expect.objectContaining({ foo: true })
    });

    fn('changeSize', [{ width: 0, height: 0 }], {
      graph: expect.objectContaining({ width: 0, height: 0 })
    });

    fn('changeMaxNoOfPoints', [{}, 10], {
      graph: expect.objectContaining({ maxNumberOfPoints: 10 })
    });

    fn('changeGraphTitle', ['title'], {
      graph: expect.objectContaining({ title: 'title' })
    });

    fn(
      'changeTicks',
      [{ minor: 1, major: 3 }],
      expect.objectContaining({
        correctResponse: [],
        graph: expect.objectContaining({ ticks: { minor: 1, major: 3 } })
      })
    );

    fn(
      'changeArrows',
      [{ left: false, right: false }],
      expect.objectContaining({
        graph: expect.objectContaining({
          arrows: { left: false, right: false }
        })
      })
    );

    describe('moveCorrectResponse', () => {});
    describe('moveInitalView', () => {});
    describe('availableTypesChange', () => {});
    describe('deleteCorrectResponse', () => {});
    describe('deleteInitialView', () => {});
    describe('addCorrectResponse', () => {});
    describe('addInitialView', () => {});
  });
});
