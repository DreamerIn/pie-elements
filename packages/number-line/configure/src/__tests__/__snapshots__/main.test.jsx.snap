// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main snapshot renders 1`] = `
<div>
  <p>
    In this interaction, students plot points, line segments or rays on a number line.
  </p>
  <FormSection
    label="Prompt"
  >
    <EditableHtml
      disableUnderline={true}
      nonEmpty={false}
      onChange={[Function]}
    />
  </FormSection>
  <WithStyles(CardBar)
    header="Attributes"
  >
    Set up the number line by entering the domain and number of tick marks to display. Labels on the number line can be edited or removed by clicking on the label.
  </WithStyles(CardBar)>
  <div>
    <FormSection
      label="Size"
    >
      <WithStyles(Size)
        onChange={[Function]}
        size={
          Object {
            "arrows": Object {
              "left": true,
              "right": true,
            },
            "availableTypes": Object {
              "LEE": true,
              "LEF": true,
              "LFE": true,
              "LFF": true,
              "PF": true,
              "REN": true,
              "REP": true,
              "RFN": true,
              "RFP": true,
            },
            "domain": Object {
              "max": 5,
              "min": -5,
            },
            "exhibitOnly": false,
            "height": 400,
            "initialElements": Array [],
            "initialType": "PF",
            "maxNumberOfPoints": 20,
            "ticks": Object {
              "major": 2,
              "minor": 1,
            },
            "width": 500,
          }
        }
      />
    </FormSection>
    <FormSection
      label="Domain"
    >
      <Domain
        domain={
          Object {
            "max": 5,
            "min": -5,
          }
        }
        onChange={[Function]}
      />
    </FormSection>
  </div>
  <div>
    <FormSection
      label="Ticks"
    >
      <WithStyles(Ticks)
        domain={
          Object {
            "max": 5,
            "min": -5,
          }
        }
        onChange={[Function]}
        ticks={
          Object {
            "major": 2,
            "minor": 1,
          }
        }
      />
    </FormSection>
    <FormSection
      label="Arrows"
    >
      <WithStyles(Arrows)
        arrows={
          Object {
            "left": true,
            "right": true,
          }
        }
        onChange={[Function]}
      />
    </FormSection>
  </div>
  <FormSection
    label="Title"
  >
    <EditableHtml
      markup=""
      onChange={[Function]}
    />
  </FormSection>
  <FormSection
    label="Limits"
  >
    <WithStyles(NumberTextField)
      label="Max No of Elements"
      max={20}
      min={1}
      onChange={[Function]}
      value={20}
    />
  </FormSection>
  <WithStyles(Button)
    color="primary"
    mini={true}
    onClick={[Function]}
    variant="outlined"
  >
    Reset to default values
  </WithStyles(Button)>
  <br />
  <br />
  <div>
    <WithStyles(CardBar)
      header="Correct Response"
    >
      Select answer type and place it on the number line. Intersecting points, line segments and/or rays will appear above the number line.
       
      <i>
        Note: A maximum of 20 points, line segments or rays may be plotted.
      </i>
    </WithStyles(CardBar)>
    <Jss(NumberLine)
      answer={Array []}
      model={
        Object {
          "correctResponse": undefined,
          "feedback": undefined,
          "graph": Object {
            "arrows": Object {
              "left": true,
              "right": true,
            },
            "availableTypes": Object {
              "LEE": true,
              "LEF": true,
              "LFE": true,
              "LFF": true,
              "PF": true,
              "REN": true,
              "REP": true,
              "RFN": true,
              "RFP": true,
            },
            "domain": Object {
              "max": 5,
              "min": -5,
            },
            "exhibitOnly": false,
            "height": 400,
            "initialElements": Array [],
            "initialType": "PF",
            "maxNumberOfPoints": 20,
            "ticks": Object {
              "major": 2,
              "minor": 1,
            },
            "title": undefined,
            "width": 500,
          },
          "prompt": undefined,
        }
      }
      onAddElement={[Function]}
      onDeleteElements={[Function]}
      onMoveElement={[Function]}
    />
    <WithStyles(CardBar)
      header="Available Types"
      mini={true}
    >
      Click on the input options to be displayed to the students. All inputs will display by default.
    </WithStyles(CardBar)>
    <div>
      <WithStyles(PointConfig)
        onSelectionChange={[Function]}
        selection={
          Object {
            "LEE": true,
            "LEF": true,
            "LFE": true,
            "LFF": true,
            "PF": true,
            "REN": true,
            "REP": true,
            "RFN": true,
            "RFP": true,
          }
        }
      />
    </div>
  </div>
  <WithStyles(CardBar)
    header="Initial view/Make Exhibit"
  >
    Use this number line to set a starting point, line segment or ray. This is optional. 
    <br />
    This number line may also be used to make an exhibit number line, which can not be manipulated by a student.
  </WithStyles(CardBar)>
  <Jss(NumberLine)
    answer={Array []}
    model={
      Object {
        "correctResponse": undefined,
        "feedback": undefined,
        "graph": Object {
          "arrows": Object {
            "left": true,
            "right": true,
          },
          "availableTypes": Object {
            "LEE": true,
            "LEF": true,
            "LFE": true,
            "LFF": true,
            "PF": true,
            "REN": true,
            "REP": true,
            "RFN": true,
            "RFP": true,
          },
          "domain": Object {
            "max": 5,
            "min": -5,
          },
          "exhibitOnly": false,
          "height": 400,
          "initialElements": Array [],
          "initialType": "PF",
          "maxNumberOfPoints": 20,
          "ticks": Object {
            "major": 2,
            "minor": 1,
          },
          "title": undefined,
          "width": 500,
        },
        "prompt": undefined,
      }
    }
    onAddElement={[Function]}
    onDeleteElements={[Function]}
    onMoveElement={[Function]}
  />
  <InputCheckbox
    checked={false}
    label="Make exhibit"
    onChange={[Function]}
    value="exhibitOnly"
  />
  <br />
  <FeedbackConfig
    onChange={[Function]}
  />
</div>
`;
